client/app/user/profile/profile.controller.js

```diff
@@ -9,23 +9,117 @@ angular.module('observatory3App')
         loggedInUser = user;
         $scope.isuser = loggedInUser._id === $stateParams.id;
         $scope.loggedInUserRole = loggedInUser.role;
-      });
 
-      $http.get('/api/users/' + $stateParams.id).success(function(user){
-          $scope.user = user;
-          $scope.originalRole = user.role;
-          $http.get('/api/commits/user/' + user.githubProfile).success(function(commits){
-              $scope.user.commits = commits;
-          });
-          // get all users projects information
-          $scope.projects = [];
-          user.projects.forEach(function(projectId){
-            $http.get("/api/projects/" + projectId).success(function(project){
-              $scope.projects.push(project);
+        //If you can access private data
+        if ($scope.isuser || $scope.loggedInUserRole === 'admin'){
+            $http.get('/api/users/' + $stateParams.id+'/private').success(function(user){
```

Comment:
also i think the private vs public data should at least call the same factory call if not the same endpoint. There is some code duplication between the private/public profile information that makes this unnecessarily long